tests/test_views.py

```diff
@@ -2821,6 +2825,36 @@ def test_claim_user_with_preprint_id_adds_corresponding_claimed_tag_to_user(self
         self.user.reload()
         assert provider_claimed_tag(self.preprint_with_source_tag.provider._id, 'preprint') in self.user.system_tags
 
+    def test_claim_user_with_registration_id_adds_corresponding_claimed_tag_to_user(self):
+        assert provider_claimed_tag(self.registration.provider._id, 'registry') not in self.user.system_tags
+        url = self.user.get_claim_url(self.registration._primary_key)
+        res = self.app.post(url, {
+            'username': self.user.username,
+            'password': 'killerqueen',
+            'password2': 'killerqueen'
+        })
+
+        assert_equal(res.status_code, 302)
+        self.user.reload()
+        assert provider_claimed_tag(self.registration.provider._id, 'registry')in self.user.system_tags
+
+    def test_claim_user_with_collected_project_id_adds_corresponding_claimed_tag_to_user(self):
+        collection_provider = CollectionProviderFactory()
+        collection = CollectionFactory(provider=collection_provider)
+        collection.collect_object(self.project, self.project.creator)
+        assert provider_claimed_tag(collection_provider._id, 'collections') not in self.user.system_tags
+
+        url = self.user.get_claim_url(self.project._primary_key)
+        res = self.app.post(url, {
+            'username': self.user.username,
+            'password': 'killerqueen',
+            'password2': 'killerqueen'
+        })
+
+        assert_equal(res.status_code, 302)
+        self.user.reload()
+        assert provider_claimed_tag(collection_provider._id, 'collections') in self.user.system_tags
+
```

Comment:
Could probably use tests for ensuring that multiple collections are tagged and that prereg, rr, and osf4m items, respectively, are all tagged along with collections.